src/WebJobs.Script/Description/FunctionDescriptorProvider.cs

```diff
@@ -72,6 +73,21 @@ public virtual bool TryCreate(FunctionMetadata functionMetadata, out FunctionDes
             }
         }
 
+        public void VerifyResolvedBindings(FunctionMetadata functionMetadata, IEnumerable<FunctionBinding> inputBindings, IEnumerable<FunctionBinding> outputBindings)
+        {
+            IEnumerable<string> bindingsFromMetadata = functionMetadata.InputBindings.Select(f => f.Type).Union(functionMetadata.OutputBindings.Select(f => f.Type));
```

Comment:
nit: this would probably be simpler if you did the same thing you've done below and union both collections, then apply the select.